### PR TITLE
Change the startup parameters of seata-server to support database-storage

### PR DESCRIPTION
**Pull request by zhangthen**
#789

### CODE REVIEW

File: server/src/main/resources/file.conf [hunk 1]
```diff
@@ -48,3 +38,31 @@ client {
...
+    user = "x"
+    password = "x"
+  }
+
+  # branch session size , if exceeded first try compress lockkey, still exceeded throws exceptions
+  max-branch-session-size = 16384
+  # globe session size , if exceeded throws exceptions
+  max-global-session-size = 512
+  # file buffer size , if exceeded allocate new buffer
+  file-write-buffer-cache-size = 16384
```
Comment (slievrly) [inline server/src/main/resources/file.conf]: should move to under the file node？also modify ConfigurationKey in the code.
image

Comment (zhangthen) [reply server/src/main/resources/file.conf]: > should move to under the file node？also modify ConfigurationKey in the code.
> image

done.

File: server/src/main/java/com/alibaba/fescar/server/session/SessionHolder.java [hunk 2]
```diff
@@ -62,24 +72,33 @@
...
-     * @param sessionStorePath the session store path
+     * @param mode the store mode: file、db
      * @throws IOException the io exception
      */
-    public static void init(String sessionStorePath) throws IOException {
-        if (sessionStorePath == null) {
-            ROOT_SESSION_MANAGER = new DefaultSessionManager(ROOT_SESSION_MANAGER_NAME);
+    public static void init(String mode) throws IOException {
+        if(StringUtils.isBlank(mode)){
+            mode = CONFIG.getConfig("store.mode");
```
Comment (slievrly) [inline server/src/main/java/com/alibaba/fescar/server/session/SessionHolder.java]: store.mode，store.file.dir-> static final, new file add copyright

Comment (zhangthen) [reply server/src/main/java/com/alibaba/fescar/server/session/SessionHolder.java]: > store.mode，store.file.dir-> static final, new file add copyright

done.

File: core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java [hunk 3]
```diff
@@ -32,6 +32,22 @@
...
+
+    /**
+     * The constant STORE_MODE.
+     */
+    public static final String STORE_MODE =  STORE_PREFIX + "mode";
+
+    /**
+     * The constant STORE_FILE_PREFIX
+     */
+    public static final String STORE_FILE_PREFIX = "file.";
```
Comment (CoffeeLatte007) [inline core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java]: lack DB_PREFIX

Comment (CoffeeLatte007) [reply core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java]: Maybe STORE_FILE_PREFIX = STORE_PREFIX + "file."?
not STORE_FILE_PREFIX = "file."

Comment (zhangthen) [reply core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java]: @CoffeeLatte007   done, exclude DB_PREFIX which is in another PR after 0.5.0

File: core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java [hunk 4]
```diff
@@ -71,5 +87,5 @@
     /**
      * The constant SERVICE_SESSION_RELOAD_READ_SIZE
      */
-    public static final String SERVICE_SESSION_RELOAD_READ_SIZE = STORE_PREFIX + "session.reload.read_size";
+    public static final String SERVICE_SESSION_RELOAD_READ_SIZE = STORE_PREFIX + STORE_FILE_PREFIX + "session.reload.read_size";
```
Comment (CoffeeLatte007) [inline core/src/main/java/com/alibaba/fescar/core/constants/ConfigurationKeys.java]: forget update  
STORE_FILE_PREFIX + "session.reload.read_size";

**Comment by CoffeeLatte007** [comment]
is's good to me . but it has conflicts

File: core/src/main/java/com/alibaba/fescar/core/store/StoreMode.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     * @param mode the mode
+     * @return the store mode
+     */
+    public static StoreMode valueof(String mode) {
+        for (StoreMode sm : values()) {
+            if (sm.name().equalsIgnoreCase(mode)) {
+                return sm;
+            }
+        }
+        return null;
```
Comment (slievrly) [inline core/src/main/java/com/alibaba/fescar/core/store/StoreMode.java]: if someone start server like `sh startup.sh 8091  /home/admin/  192.168.0.1` ,I think it should be throw IllegalArgumentException, not return null and use default file mode.

File: server/src/test/java/com/alibaba/fescar/server/store/SessionStoreTest.java [hunk 6]
```diff
@@ -69,7 +69,8 @@ public void clean() throws Exception {
      */
     @Test
     public void testRestoredFromFile() throws Exception {
-        SessionHolder.init(".");
+//        SessionHolder.init(".");
```
Comment (slievrly) [inline server/src/test/java/com/alibaba/fescar/server/store/SessionStoreTest.java]: directly delete

Comment (zhangthen) [reply server/src/test/java/com/alibaba/fescar/server/store/SessionStoreTest.java]: > directly delete

done.

**Pull request merged**